### DISCUSSION TITLE
How to use HTTPS requests instead of HTTP

### DISCUSSION THREAD

**Discussion by ithuangzeming** [mpromonet/webrtc-streamer#618] [Q&A]
My web page is an HTTPS request,but my request for webrtc is an HTTP request.
The following error will be reported:
This request has been blocked; the content must be served over HTTPS.

**Comment by mpromonet** [comment]
Hi,
You have several choices:
- proxyfying https to http
- enable https on webrtc-streamer

Best Regards,
Michel.

**Reply by ithuangzeming** [reply]
How to enable https on webrtc-streamer,could you give an example

**Comment by mpromonet** [comment]
Hi,

See https://github.com/civetweb/civetweb/blob/master/docs/OpenSSL.md
Adding arguments `-H8443s -c sslkeycert.pem`

Best Regards,
Michel.

**Reply by ithuangzeming** [reply]
Follow the command below:
webrtc-streamer.exe -H8443s 192.168.100.121 -c 9179525_console.test.qyiot.top.pem
An error occurred:
Initializing SSL failed: -ssl_certificate is not set
Error initializing SSL context
Cannot Initialize start HTTP server exception:null context when constructing CivetServer. Possible problem binding to port. Error: Error initializing SSL context
Exit

**Comment by ithuangzeming** [comment]
Follow the command below:
webrtc-streamer.exe -H8443s 192.168.100.121 -c 9179525_console.test.qyiot.top.pem
An error occurred:
Initializing SSL failed: -ssl_certificate is not set
Error initializing SSL context
Cannot Initialize start HTTP server exception:null context when constructing CivetServer. Possible problem binding to port. Error: Error initializing SSL context
Exit

**Reply by mpromonet** [reply]
Did you put Key and certificate in the pem file ?

**Reply by mpromonet** [reply]
I upgrade ssl api used, it seems to fix the problem https://github.com/mpromonet/webrtc-streamer/commit/81b641f33553ce6af1cc78e1ca9f4c8215124fef

**Reply by ithuangzeming** [reply]
Still not working:
Same error reported.
This is my directory
![image](https://github.com/mpromonet/webrtc-streamer/assets/53346164/9b854354-d7b1-48ea-95d1-62c5a4024b3d)

**Reply by ithuangzeming** [reply]
![image](https://github.com/mpromonet/webrtc-streamer/assets/53346164/7b970938-6a54-4155-a98c-173899fe6a47)

**Reply by mpromonet** [reply]
It should not be a directory, it need to be a file that contains both content key & certificate

**Reply by ithuangzeming** [reply]
I succeeded, thank you

**Reply by ithuangzeming** [reply]
but my another PC has occurred error :
@mpromonet 
initialize_openssl: error loading library libcrypto-1_1-x64.dll

**Reply by ithuangzeming** [reply]
Is there any solution for the error mentioned last time?

**Reply by mpromonet** [reply]
Does the library exist ?

**Reply by ithuangzeming** [reply]
But before, my computer didn't have a library and it was running normally @mpromonet

**Reply by ithuangzeming** [reply]
@mpromonet It can be used normally now,using the latest version

**Comment by 17-phper** [comment]
@mpromonet  May I ask the following。When I start an https，it receded
![image](https://github.com/mpromonet/webrtc-streamer/assets/40785229/e4edb0e8-3250-46b7-8c7e-16614e5f55fc)

**Reply by 17-phper** [reply]
@mpromonet 
I have already put Key and certificate in the pem file
It exits with no error message

**Reply by mpromonet** [reply]
v0.6.4 is quite old ...

**Reply by testaaa456** [reply]
> @mpromonet I have already put Key and certificate in the pem file It exits with no error message
------------

@mpromonet  hi, I have the same "error" , it exits without any message as below..  help...

![webrtc](https://github.com/user-attachments/assets/df3fac97-0939-4655-8fa5-1e0c44f87f8b)

**Reply by mpromonet** [reply]
Hi @testaaa456,
from the help
![image](https://github.com/user-attachments/assets/29efeb91-22a1-4407-8b2e-07b62e3d2f59)
You should better try `-H0.0.0.0:8443s`
Best Regards,
Michel.

**Reply by testaaa456** [reply]
> Hi @testaaa456, from the help ![image](https://private-user-images.githubusercontent.com/1367630/375950034-29efeb91-22a1-4407-8b2e-07b62e3d2f59.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjg5MDU1MDIsIm5iZiI6MTcyODkwNTIwMiwicGF0aCI6Ii8xMzY3NjMwLzM3NTk1MDAzNC0yOWVmZWI5MS0yMmExLTQ0MDctOGIyZS0wN2I2MmUzZDJmNTkucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTAxNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEwMTRUMTEyNjQyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YTQ4NjE2Njg5YzM5ZTdlODIyOTRmOTc3MmJmNTc1NmEwYTEwNTI4YjQyZDJmMTI2MzYyN2YxNWUwN2VjZDA5ZiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.6nfcpt8Jatlz6yV4RltmbHh-_tk3xoOgY00SDGl9fww) You should better try `-H0.0.0.0:8443s` Best Regards, Michel.

![图片](https://github.com/user-attachments/assets/1f6a316b-0be2-4d03-8519-e647fe75e191)

hi @mpromonet ,  still have the same situtation,   is there something wrong with my pem file ?
